src/js.rs

```diff
@@ -11,6 +11,85 @@ impl Default for JsGenerator {
     }
 }
 
+#[derive(Default)]
+pub struct TsGenerator;
+
+impl TsGenerator {
+    pub fn generate(&self, iface: Interface) -> js::Tokens {
+        // TODO: Box Type
```

Comment:
the only public api on box would be drop.

Reply:
do we need to expose it in ts?

Reply:
would say so yes. while it's definitely better to define an object, there is support for returning raw boxes. Although we could remove it. it was a stepping stone to building objects. it's really unsafe to pass a `Box<B>` where an `&A` or `Box<A>` is expected. we could build in a check, but that's probably not worth it.

Reply:
yeah, let's remove this eventually, opened an issue for it. It's a learning process to figure out what can be exposed safely, and the Box is a useful internal abstraction but it shouldn't leak to the user.